### DISCUSSION TITLE
key-value类配置和结构化数据配置探讨

### DISCUSSION THREAD

**Discussion by yearsFung** [apolloconfig/apollo#4115] [Q&A]
如题，在用apollo配置过程中联想到对配置数据建模的问题。
       apollo只支持key-value这样通用型的键值对配置，这样的好处是足够通用，不需随着业务演化而变化。也确实不应该要求apollo通用型的框架去支持业务属性重的结构化数据配置，因为业务上就行不通，也无法做成通用型框架。
       对于结构化、需要随着业务演化而变化的的配置数据，我认为应该按照DDD的思想进行领域划分，作为交易系统的一部分，而不应该把交易系统的配置独立“拿出来”作为单独的配置系统存在，因为这样不贴合业务系统模型，无法随着业务进行合理的演进。但公司目前有一种声音是认为要把所有“带人机交互界面”的配置全部交由专门的运营开发团队负责，从而达到在运营端统一透出的目的，甚至是减轻交易系统开发团队的工作量，这样是否合理？我感觉是不合理的。
       综上，我想到的结论或解决方案是：1、通用型key-value 统一配置、统一维护，由类似apollo这样的通用系统处理；2、结构化的配置（业务属性重的）按DDD思想建模，本身配置就是与交易系统紧耦合的，需要统一透出可以通过单点登录+统一认证实现。请问老师的意见呢？
       thx！

**Comment by Anilople** [comment]
Is there an example about it?
